json/src/main/scala/com/gu/contentapi/json/CirceSerialization.scala

```diff
@@ -0,0 +1,177 @@
+  implicit val blocksDecoder = genBlocksDecoder
+  implicit val rightsDecoder = Decoder[Rights]
+  implicit val crosswordEntryDecoder = genCrosswordEntryDecoder
+  implicit val crosswordDecoder = Decoder[Crossword]
+  implicit val contentStatsDecoder = Decoder[ContentStats]
+  implicit val sectionDecoder = Decoder[Section]
+  implicit val debugDecoder = Decoder[Debug]
+  implicit val contentDecoder = Decoder[Content]
+
+  // These two need to be written manually. I think the `Map[K, V]` type having 2 type params causes implicit divergence,
+  // although shapeless's Lazy is supposed to work around that.
+
+  def genBlocksDecoder(implicit blockDecoder: Decoder[Block]): Decoder[Blocks] = Decoder.instance[Blocks] { cursor =>
+    for {
+      main <- cursor.get[Option[Block]]("main")
+      body <- cursor.get[Option[Seq[Block]]]("body")
+      totalBodyBlocks <- cursor.get[Option[Int]]("totalBodyBlocks")
+      requestedBodyBlocks <- cursor.get[Option[Map[String, Seq[Block]]]]("requestedBodyBlocks")
+    } yield Blocks(main, body, totalBodyBlocks, requestedBodyBlocks)
+  }
+
+  def genCrosswordEntryDecoder(implicit dec: Decoder[Option[Map[String,Seq[Int]]]]): Decoder[CrosswordEntry] = Decoder.instance[CrosswordEntry] { cursor =>
+    for {
+      id <- cursor.get[String]("id")
+      number <- cursor.get[Option[Int]]("number")
+      humanNumber <- cursor.get[Option[String]]("humanNumber")
+      direction <- cursor.get[Option[String]]("direction")
+      position <- cursor.get[Option[CrosswordPosition]]("position")
+      separatorLocations <- cursor.get[Option[Map[String, Seq[Int]]]]("separatorLocations")
+      length <- cursor.get[Option[Int]]("length")
+      clue <- cursor.get[Option[String]]("clue")
+      group <- cursor.get[Option[Seq[String]]]("group")
+      solution <- cursor.get[Option[String]]("solution")
+      format <- cursor.get[Option[String]]("format")
+    } yield CrosswordEntry(
+      id,
+      number,
+      humanNumber,
+      direction,
+      position,
+      separatorLocations,
+      length,
+      clue,
+      group,
+      solution,
+      format
+    )
+  }
+
+  object AtomDeserialization {
+
+    implicit val decodeUnknownOpt: Decoder[AtomData.UnknownUnionField] =
+      Decoder.instance(c =>
+        Xor.left(DecodingFailure("AtomData.UnknownUnionField", c.history))
+      )
+
+    def getAtom(c: HCursor): Decoder.Result[Atom] = {
+      for {
+        atomType <- c.downField("atomType").as[AtomType]
+        atomData <- getAtomData(c, atomType)
+        atom <- getAtom(c, atomData)
+      } yield atom
+    }
+
+    def getAtoms(c: HCursor): Decoder.Result[Atoms] = {
+      for {
+        quizzes <- getAtoms(c, AtomType.Quiz)
+        media <- getAtoms(c, AtomType.Media)
+      } yield {
+        Atoms(quizzes = quizzes, media = media)
+      }
+    }
+
+    private def getAtom(c: HCursor, atomData: AtomData): Decoder.Result[Atom] = {
+      for {
+        id <- c.downField("id").as[String]
```

Comment:
`c.downField("foo").as[Bar]` can be written slightly more simply as `c.get[Bar]("foo")`